packages/@ember/-internals/utils/lib/cache.ts

```diff
@@ -1,21 +1,26 @@
-export default class Cache<T, V> {
+export default class Cache<
+  T,
+  V extends object | number | string | boolean | bigint | symbol | null
```

Comment:
Can you add a comment explaining why this is essentially any type except `undefined`? This is a common technique/optimization when using Maps so it should be apparent, but a comment would help prevent anyone from accidentally removing this optimization in the future.